rtengine/ipwavelet.cc

```diff
@@ -2216,67 +2221,61 @@ void ImProcFunctions::ip_wavelet(LabImage * lab, LabImage * dst, int kall, const
-                        float iny = log(rapX);
-                        float rap = inx / iny; //koef
-                        float asig = 0.166f / (tempsig * sig);
-                        float bsig = 0.5f - asig * tempmean;
-                        float amean = 0.5f / (tempmean);
-                    
+                        constexpr float insigma = 0.666f; //SD
+                        const float logmax = log(tempmax); //log Max
+                        const float rapX = (tempmean + sig * tempsig) / tempmax; //rapport between sD / max
+                        const float inx = log(insigma);
+                        const float iny = log(rapX);
+                        const float rap = inx / iny; //koef
+                        const float asig = 0.166f / (tempsig * sig);
+                        const float bsig = 0.5f - asig * tempmean;
+                        const float amean = 1.f / tempmean;
 
 #ifdef _OPENMP
         #pragma omp parallel for schedule(dynamic, Wlvl_L * 16) num_threads(wavNestedLevels) if (wavNestedLevels>1)
 #endif
 
                         for (int i = 0; i < Wlvl_L * Hlvl_L; i++) {
                             float absciss;
-                            float tempwav = 0.f;
-                            tempwav = 0.7f * WavCoeffs_L[dir][i] + 0.3f * WavCoeffs_L2[dir][i];
+                            const float tempwav = std::fabs(0.7f * WavCoeffs_L[dir][i] + 0.3f * WavCoeffs_L2[dir][i]);
 
-                            if (std::fabs(tempwav) >= (tempmean + sig * tempsig)) { //for max
-                                float valcour = xlogf(std::fabs(tempwav));
-                                float valc = valcour - logmax;
-                                float vald = valc * rap;
+                            if (tempwav >= tempmean + sig * tempsig) { //for max
+                                const float vald = (xlogf(tempwav) - logmax) * rap;
                                 absciss = xexpf(vald);
-                            } else if (std::fabs(tempwav) >= tempmean) {
-                                absciss = asig * std::fabs(tempwav) + bsig;
+                            } else if (tempwav >= tempmean) {
+                                absciss = asig * tempwav + bsig;
                             } else {
-                                absciss = amean * std::fabs(tempwav);
-                                float abs = pow(2.f * absciss, (1.f / SQR(sig)));
-                                absciss = 0.5f * abs;
-                            }
-                            float kc = wavguid.getVal(absciss) -1.f;
-
-                            if(kc < 0) {
-                                kc = -SQR(kc);//approximation to simulate sliders denoise
+                                absciss = amean * tempwav;
+                                /*
+                                if (sig == 2.f) { // for sig = 2.f we can use a faster calculation because the exponent in this case is 0.25
```

Comment:
If we go for commented-out code, I would prefer to have a static assert here to check the value of `sig`.

Reply:
Thinking about this some, we really should not do this: The comments are not compiled and will just rot away. Having the slow path will not affect the fast path at all as it should be removed by the optimizer. I think the the initial approach posted was superior.

Reply:
I agree about comment-rot, we have a lot of that tbh. If anyone says/proves that the old, more verbose way, does not results in a significant slowdown, it should be kept. My argument would be that a lot of algorithms we use are not so well documented. Doing 'smart' optimizations in the code might lead to semantic-rot, which is worse imo.

Reply:
a static assert of a constexpr?

> If anyone says/proves that the old, more verbose, way does not result in a significant slowdown, it should be kept.

So I will keep it. The only downside is, that the file now is not cppcheck clean.